src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -270,8 +274,31 @@ def schedule_task(self, task):
         if not cores_per_slot:
             cores_per_slot = 1
 
-        self._log.debug_3('req : %s %s %s %s %s', req_slots, cores_per_slot,
-                        gpus_per_slot, lfs_per_slot, mem_per_slot)
+        # check if there is a GPU sharing
+        if gpus_per_slot and not gpus_per_slot.is_integer():
+            gpus           = td['ranks'] * gpus_per_slot
+            req_slots      = m.ceil(gpus / gpus_per_node)
+            gpus_per_slot  = int(gpus // req_slots)
+            ranks_per_slot = td['ranks'] // req_slots
```

Comment:
Ok, understood that a `slot` is not something which can hold a number of ranks so that all GPUs on those ranks are included.

But I don't think the math works out:
```py
import math as m

td = {'ranks'         : 9,
      'cores_per_rank': 1,
      'gpus_per_rank' : 1 / 3}

cores_per_node = 128
gpus_per_node  =   2


# from `schedule_task()`
cores_per_slot = td['cores_per_rank']
gpus_per_slot  = td['gpus_per_rank']

gpus           = td['ranks'] * gpus_per_slot
req_slots      = m.ceil(gpus / gpus_per_node)
gpus_per_slot  = int(gpus // req_slots)
ranks_per_slot = td['ranks'] // req_slots

# find the greatest common divisor
di = m.gcd(gpus_per_slot, ranks_per_slot)
req_slots       *= di
gpus_per_slot  //= di
ranks_per_slot //= di

print('req_slots     : ', req_slots)
print('ranks_per_slot: ', ranks_per_slot)
print('gpus_per_slot : ', gpus_per_slot)
```

which gives 
```sh
$ ./t.py
req_slots     :  2
ranks_per_slot:  4
gpus_per_slot :  1
```

I would have expected 
```sh
$ ./t.py
req_slots     :  3
ranks_per_slot:  3
gpus_per_slot :  1
```

Am I missing something?

Reply:
yeah, need to polish this part, thank you for catching it

Reply:
resolved